app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/DeviceListFragment.java

```diff
@@ -0,0 +1,357 @@
+
+      default: return getString(R.string.unknown);
+    }
+  }
+
+  /* From WifiP2pManager.PeerListListener callback-interface */
+  @Override
+  public void onPeersAvailable(WifiP2pDeviceList peers) {
+
+    ProgressBar searchingPeersProgressBar = deviceListFragmentRootView.findViewById(R.id.progress_bar_searching_peers);
+    searchingPeersProgressBar.setVisibility(View.GONE);
+    FrameLayout frameLayoutPeerDevices = deviceListFragmentRootView.findViewById(R.id.frame_layout_peer_devices);
+    frameLayoutPeerDevices.setVisibility(View.VISIBLE);
+
+    peerDevices.clear();
+    peerDevices.addAll(peers.getDeviceList());
+    ((WifiPeerListAdapter) getListAdapter()).notifyDataSetChanged();
+
+    if(peerDevices.size() == 0) {
+      Log.d(LocalFileTransferActivity.TAG, "No devices found");
+    }
+  }
+
+  public void clearPeers() {
+    peerDevices.clear();
+    ((WifiPeerListAdapter) getListAdapter()).notifyDataSetChanged();
+  }
+
+  public void onInitiateDiscovery() { // Setup UI for searching peers
+    ProgressBar searchingPeersProgressBar = deviceListFragmentRootView.findViewById(R.id.progress_bar_searching_peers);
+    searchingPeersProgressBar.setVisibility(View.VISIBLE);
+    FrameLayout frameLayoutPeerDevices = deviceListFragmentRootView.findViewById(R.id.frame_layout_peer_devices);
+    frameLayoutPeerDevices.setVisibility(View.INVISIBLE);
+  }
+
+  /* From WifiP2pManager.ConnectionInfoListener callback-interface */
+  @Override
+  public void onConnectionInfoAvailable(WifiP2pInfo info) {
+    /* Devices have successfully connected, and 'info' holds information about the wifi p2p group formed */
+    groupInfo = info;
+    // Start handshake between the devices
+    new PeerGroupHandshakeAsyncTask(this, groupInfo).execute();
+  }
+
+  public void setClientAddress(InetAddress clientAddress) {
+    if(clientAddress == null) {
+      // null is returned only in case of a failed handshake
+      showToast(localFileTransferActivity, R.string.device_not_cooperating, Toast.LENGTH_LONG);
+      localFileTransferActivity.closeLocalFileTransferActivity();
+      return;
+    }
+
+    // If control reaches here, means handshake was successful
+    selectedPeerDeviceInetAddress = clientAddress;
+    startFileTransfer();
+  }
+
+  private void startFileTransfer() {
+    fileTransferStarted = true;
+
+    if(groupInfo.groupFormed && !fileSender) {
+      displayTransferProgressFragment();
+
+      new ReceiverDeviceAsyncTask(this, transferProgressFragment).execute();
+      showToast(localFileTransferActivity, R.string.preparing_to_receive, Toast.LENGTH_SHORT);
+
+    } else if(groupInfo.groupFormed) {
+      {
+        Log.d(LocalFileTransferActivity.TAG, "Starting file transfer");
+
+        showToast(localFileTransferActivity, R.string.starting_transfer, Toast.LENGTH_SHORT);
+
+        if(groupInfo.isGroupOwner)  fileReceiverDeviceAddress = selectedPeerDeviceInetAddress;
```

Comment:
should be a ternary

Reply:
At this rate I'll probably take an eternity to internalise using ternary :man_facepalming:

Reply:
heh, no worries, once you get in the habit they are just the best

Reply:
[Done](https://github.com/kiwix/kiwix-android/pull/1276/commits/c6de4045c87b493399f574679e8b26614ed15132#diff-c2e372482ef0816a999e44dec2c63c52R239)